crates/autopilot/src/solvable_orders.rs

```diff
@@ -236,6 +249,29 @@ impl SolvableOrdersCache {
     pub fn last_update_time(&self) -> Instant {
         self.cache.lock().unwrap().orders.update_time
     }
+
+    /// Filter out limit orders which are far enough outside the estimated native token price.
+    fn filter_mispriced_limit_orders(
+        &self,
+        mut orders: Vec<Order>,
+        prices: &BTreeMap<H160, U256>,
+    ) -> Vec<Order> {
+        orders.retain(|order| {
+            if order.metadata.class != OrderClass::Limit {
+                return true;
+            }
+
+            // Convert the sell and buy price to the native token (ETH) and make sure that sell
+            // with the surplus fee is higher than buy with the configurable price factor.
+            let sell_native = order.data.sell_amount * prices.get(&order.data.sell_token).unwrap()
+                + order.metadata.surplus_fee;
```

Comment:
Is it correct that surplus_fee is not multiplied by the price? Is the unit in the sell token or native token? Maybe we should comment that on to OrderMetadata::surplus_fee .